src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
```

Comment:
suggest to use reduce method instead forEach and immediately return result of reduce
```
return items.reduce((groupItems: GroupsMap<T>, item) => {
   do something
})
```